io/src/test/scala/sbt/internal/io/SourceModificationWatchSpec.scala

```diff
@@ -0,0 +1,296 @@
+package sbt.internal.io
+
+import java.nio.file.{ ClosedWatchServiceException, Paths }
+
+import org.scalatest.{ Assertion, FlatSpec, Matchers }
+import sbt.io.syntax._
+import sbt.io.{ IO, SimpleFilter, WatchService }
+
+import scala.concurrent.duration._
+
+abstract class SourceModificationWatchSpec(getService: => WatchService, pollDelay: FiniteDuration, maxWait: FiniteDuration) extends FlatSpec with Matchers {
```

Comment:
is this tested on all three major platforms? Because the semantics are very different and there are OS limitations to fidelity of changes (e.g. the Linux kernel only records FS accuracy to second precision on most filesystems)